src/devices/Card/Ultralight/README.md

```diff
@@ -0,0 +1,159 @@
+# Ultralight card - RFID Card
+
+This class supports Ultralight cards. They are RFID cards responding to ISO 14443 type A. You need a specific card reader like PN532, PN5180 or MFRC522 to read, write those kind of cards.
+
+## Creating a card and reading it
+
+You will find detailed examples for PN532 [here](../../Pn532/samples), for MFRC522 [here](../../Mfrc522/samples) and for PN5180 [here](../../Pn5180/samples).
+
+You'll need first to get the card from an RFID reader. The example below shows how to do it with a MFRC522 and read all the sectors, read the configuration data, and print all the sector data information, read NDEF, write NDEF.
+
+```csharp
+var ultralight = new UltralightCard(mfrc522!, 0);
+    ultralight.SerialNumber = card.NfcId;
+    Console.WriteLine($"Type: {ultralight.UltralightCardType}, Ndef capacity: {ultralight.NdefCapacity}");
+
+    var version = ultralight.GetVersion();
+    if ((version != null) && (version.Length > 0))
+    {
+        Console.WriteLine("Get Version details: ");
+        for (int i = 0; i < version.Length; i++)
+        {
+            Console.Write($"{version[i]:X2} ");
+        }
+
+        Console.WriteLine();
+    }
+    else
+    {
+        Console.WriteLine("Can't read the version.");
+    }
+
+    var sign = ultralight.GetSignature();
+    if ((sign != null) && (sign.Length > 0))
+    {
+        Console.WriteLine("Signature: ");
+        for (int i = 0; i < sign.Length; i++)
+        {
+            Console.Write($"{sign[i]:X2} ");
+        }
+
+        Console.WriteLine();
+    }
+    else
+    {
+        Console.WriteLine("Can't read the signature.");
+    }
+
+    // The ReadFast feature can be used as well, note that the MFRC522 has a very limited FIFO
+    // So maximum 9 pages can be read as once.
+    Console.WriteLine("Fast read example:");
+    var buff = ultralight.ReadFast(0, 8);
+    if (buff != null)
+    {
+        for (int i = 0; i < buff.Length / 4; i++)
+        {
+            Console.WriteLine($"  Block {i} - {buff[i * 4]:X2} {buff[i * 4 + 1]:X2} {buff[i * 4 + 2]:X2} {buff[i * 4 + 3]:X2}");
+        }
+    }
+
+    Console.WriteLine("Dump of all the card:");
+    for (int block = 0; block < ultralight.NumberBlocks; block++)
+    {
+        ultralight.BlockNumber = (byte)block; // Safe cast, can't be more than 255
+        ultralight.Command = UltralightCommand.Read16Bytes;
+        var res = ultralight.RunUltralightCommand();
+        if (res > 0)
+        {
+            Console.Write($"  Block: {ultralight.BlockNumber:X2} - ");
+            for (int i = 0; i < 4; i++)
+            {
+                Console.Write($"{ultralight.Data![i]:X2} ");
```

Comment:
consider making `Data` non-nullable (i.e. return empty array) and fix sample so it doesn't have any nullable related `!`

Reply:
OK, will do and as well in the MifareCard